archivist/sboms.py

```diff
@@ -66,12 +69,149 @@ def __init__(self, archivist: "type_helper.Archivist"):
     def __str__(self) -> str:
         return f"SBOMSClient({self._archivist.url})"
 
+    @staticmethod
+    def parse(data: Dict) -> Dict:  # pragma: no cover
+        """
+        parse the sbom and extract pertinent informtion
+
+        Args:
+            data (dict): dictionary
+
+        A YAML representation of the data argument would be:
+
+            .. code-block:: yaml
+
+                filename: functests/test_resources/sboms/gen1.xml
+
+            OR
+
+            .. code-block:: yaml
+
+                url: https://some.hostname/cdx.xml
+
+             Either 'filename' or 'url' is required.
+
+        Returns:
+
+            A dict suitable for adding to an asset or event creation
+
+        """
+        result = None
+        filename = data.get("filename")
+        if filename is not None:
+            with open(filename, "rb") as fd:
+                sbom = xmltodict_parse(fd, xml_attribs=True, disable_entities=False)
+
+        else:
+            url = data["url"]
+            fd = BytesIO()
+            get_url(url, fd)
+            sbom = xmltodict_parse(fd, xml_attribs=True, disable_entities=False)
+
+        b = sbom["bom"]
+        m = b["metadata"]
+        c = m["component"]
+        hash_value = c["hashes"]["hash"]["#text"]
+        result = {
+            "author": c["author"],
+            "component": c["name"],
+            "hash": hash_value,
+            "repo": "",  # unsure about this
```

Comment:
repository is currently not reliable in SBOM specs. Suggest leaving it blank for now, can be patched later if someone comes to depend on it.